sqa/tron/query.py

```diff
@@ -0,0 +1,451 @@
+from typing import TypedDict, Iterable
+
+import marshmallow as mm
+from pyarrow.dataset import Expression
+
+from sqa.query.model import JoinRel, RefRel, Table, Item, FieldSelection, Scan
+from sqa.query.schema import BaseQuerySchema, field_map_schema
+from sqa.query.util import to_snake_case, json_project, get_selected_fields, field_in
+
+
+class BlockFieldSelection(TypedDict, total=False):
+    number: bool
+    hash: bool
+    parentHash: bool
+    timestamp: bool
+    txTrieRoot: bool
+    version: bool
+    timestamp: bool
+    witness_address: bool
+    witness_signature: bool
+
+
+TxFieldSelection = TypedDict('TxFieldSelection', {
+    'hash': bool,
+    'ret': bool,
+    'signature': bool,
+    'type': bool,
+    'parameter': bool,
+    'permissionId': bool,
+    'refBlockBytes': bool,
+    'refBlockHash': bool,
+    'feeLimit': bool,
+    'expiration': bool,
+    'timestamp': bool,
+    'rawDataHex': bool,
+    'fee': bool,
+    'contractResult': bool,
+    'contractAddress': bool,
+    'resMessage': bool,
+    'withdrawAmount': bool,
+    'unfreezeAmount': bool,
+    'withdrawExpireAmount': bool,
+    'cancelUnfreezeV2Amount': bool,
+    'result': bool,
+    'energyFee': bool,
+    'energyUsage': bool,
+    'energyUsageTotal': bool,
+    'netUsage': bool,
+    'netFee': bool,
+    'originEnergyUsage': bool,
+    'energyPenaltyTotal': bool,
+}, total=False)
+
+
+class LogFieldSelection(TypedDict, total=False):
+    logIndex: bool
```

Comment:
Let's remove it to match rest networks. `transaction_index` should be added as a non-key field.